### ISSUE TITLE
[Raid] Sunwell Plateau - Kalecgos

### ISSUE THREAD

**Issue by NightSeekerr** [WoWManiaUK/Redemption#2768]
Problems with 2 boss fights. 

Kalecgos (1st boss fight): Once the boss is about to die he becomes immune to dmg and cannot be killed. I had to call GM to kill it both times I did Sunwell Plateau and I know for a fact it's a known issue among the GMs. That said, I don't know how the fight should work exactly, other than the boss being killable. Boss link: https://www.wowhead.com/npc=24850/kalecgos

Brutallus: After Brutallus is defeated something bugs and the raid group cannot proceed any further (there's a wall of fire surrounding the boss area), so again both times I did that instance I had to call a GM for help with that issue too. Boss link: https://www.wowhead.com/npc=24882/brutallus

I use the launcher and latest client.

**Comment by wow-mania** [comment]
We need specific steps to reproduce this.

**Comment by Dranah** [comment]
Ok this is a bit more complex than that.
I can confirm that the fight is currently not doable, as both the reset mechanisms, both the kill credit mechanism is missing.

**How it should work:**
Kalecgos is not supposed to "die" as he is not your enemy, he is misguided and fights for control against the demon Sathrovarr. Sathrovarr is the one in the Spectral realm, fighting Kalec's human form.
Whenever one of them is about to die, at 1% health they are banished and the fight only ends when they are both at 1% health, banished.
If you succeed, it is Sathrovarr (who is in the spectral realm) who dies and his corpse spawns up on the platform to loot, while Kalecgos turns friendly.

However, there are many ways to fail and several reset mechanisms for those cases. Listing:
1) If Kalec dies in the Spectral Realm, the fight is supposed to force despawn, Kalecgos disappears for 20sec and every raid member is teleported back into the normal realm, out of combat if alive.
2) If Kalecgos in normal Realm has no aggro target, fight should force despawn, Kalecgos disappears for 20sec everyone is back to normal realm etc, then he respawns.

**How it works atm:**
- If you do the fight as intended, Kalec is alive, Sathrovarr is banished at 1%, Kalecgos is banished at 1%, it simply doesn't end the fight, nothing happens.
- If Kalec dies in the Spectral realm, the fight does not reset. This becomes an issue when your raid already banished Kalecgos and he stops spawning more portals, at this point the people in the Spectral realm have to finish off Sathrovarr before their time in the realm expires. If they fail to do so, Kalec remains alone and soon dies, which would normally end the combat, but here it doesn't. Fight gets stuck.
- There is a problem with getting double spawn of Sathrovarr in the Spectral realm. There is just 2 of him for no apparent reason.

TL;DR version: whatever you do, the fight gets stuck.

**Comment by Keader** [comment]
@Dranah any news testing this?

**Comment by Keader** [comment]
@Dranah Kalecgos : I ported fix of Trinity (made by sirikfoll) after 3 hours ._.
About Brutallus, no one wants fix full event, is sucks.
But in pratical terms, in code felmyst is summoned when brutallus die.
So... right now (is wrong i know), he should be summoned when brutallus die, without any problem caused by corpse.
So i cant see how it will not work.

**Comment by Dranah** [comment]
UPDATE:
- Sathrovarr double spawn issue fixed (fixed)
- Encounter properly ends now when both targets are banished (fixed)

New issues:

1) Spectral Exhaustion (https://www.wowhead.com/spell=44867/spectral-exhaustion) effect fails to prevent Kalecgos from casting Spectral Blast on you, teleporting you to the shadow realm even though you have the debuff (need fix)
2) Sathrovarr continues casting https://www.wowhead.com/spell=45032/curse-of-boundless-agony while banished (need fix)
3) Players being transfered back from shadow realm:
Dead players currently don't get teleported back from the shadow realm to the normal realm when the encounter ends any way (need fix)
Living players get teleported back properly if the encounter is force reset by a wipe or otherwise if their debuff expires, however they should also get teleported back instantly when the encounter ends (need fix)
4) Visibility issue with Sathrovarr's corpse when the encounter is defeated: only players who were in the shadow realm when the fight ended are able to see it, players who were in the normal phase can not.
5) Kalec in the shadow realm is wrongly named as "Kalecgos" when his name should be just "Kalec" in human form (minor issue)

**Comment by Keader** [comment]
What bugs remain here @Dranah ?

**Comment by Dranah** [comment]
Was that priority label really needed? -_-

Ontopic:
- Spectral exhaustion debuff now prevents players from porting to the spectral realm by any means
- Sathrovarr is no longer casting Curse of Boundless Agony after he is already banished and he no longer targets players in the other realm
- Living players in the spectral realm are now teleported back to the normal realm when the encounter ends by any means
- Player corpses are teleported back right away when they die
- Sathrovarr's corpse is now visible to players regardless of which realm they were in
- The encounter properly ends when the criteria is met, Kalecgos turns friendly and Sathrovarr is lootable
- The encounter is properly reset when there are no players alive in the normal realm or if the NPC dies in spectral realm

Reset time is a little too quick, players may not have enough time to leave the pull range after a force-reset.
How it should work: Kalecgos should only respawn 15-20sec after if a reset happens.

**Comment by Dranah** [comment]
UPDATE:
- Reset time is now proper 20sec
